Compiler/src/abstractinterpretation.jl

```diff
@@ -3144,7 +3135,7 @@ function abstract_eval_splatnew(interp::AbstractInterpreter, e::Expr, sstate::St
                     all(i::Int -> ⊑(𝕃ᵢ, (at.fields::Vector{Any})[i], fieldtype(t, i)), 1:n)
                 end))
             nothrow = isexact
-            rt = PartialStruct(𝕃ᵢ, rt, at.fields::Vector{Any})
+            rt = PartialStruct(𝕃ᵢ, rt, at.undef, at.fields::Vector{Any})
```

Comment:
Since `:splatnew` iterates through every field, even if the `PartialStruct` at `at` might have undefined fields, we can be sure there won't be any undefined fields after `:splatnew` runs successfully.
```suggestion
            rt = PartialStruct(𝕃ᵢ, rt, trues(length(at.undef)), at.fields::Vector{Any})
```